cmd/podman/service.go

```
@@ -0,0 +1,149 @@
+	"github.com/containers/libpod/pkg/adapter"
+	api "github.com/containers/libpod/pkg/api/server"
+	"github.com/containers/libpod/pkg/rootless"
+	"github.com/containers/libpod/pkg/util"
+	"github.com/containers/libpod/pkg/varlinkapi"
+	"github.com/containers/libpod/version"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/varlink/go/varlink"
+)
+
+var (
+	serviceCommand     cliconfig.ServiceValues
+	serviceDescription = `Run an API service
+
+Enable a listening service for API access to Podman commands.
+`
+
+	_serviceCommand = &cobra.Command{
+		Use:   "service [flags] [URI]",
+		Short: "Run API service",
+		Long:  serviceDescription,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			serviceCommand.InputArgs = args
+			serviceCommand.GlobalFlags = MainGlobalOpts
+			return serviceCmd(&serviceCommand)
+		},
+	}
+)
+
+func init() {
+	serviceCommand.Command = _serviceCommand
+	serviceCommand.SetHelpTemplate(HelpTemplate())
+	serviceCommand.SetUsageTemplate(UsageTemplate())
+	flags := serviceCommand.Flags()
+	flags.Int64VarP(&serviceCommand.Timeout, "timeout", "t", 1000, "Time until the service session expires in milliseconds.  Use 0 to disable the timeout")
+	flags.BoolVar(&serviceCommand.Varlink, "varlink", false, "Use legacy varlink service instead of REST")
+}
+
+func serviceCmd(c *cliconfig.ServiceValues) error {
+	// For V2, default to the REST socket
+	apiURI := adapter.DefaultAPIAddress
+	if c.Varlink {
+		apiURI = adapter.DefaultVarlinkAddress
+	}
+
+	if rootless.IsRootless() {
+		xdg, err := util.GetRuntimeDir()
+		if err != nil {
+			return err
+		}
+		socketName := "podman.sock"
+		if c.Varlink {
+			socketName = "io.podman"
+		}
+		socketDir := filepath.Join(xdg, "podman", socketName)
+		if _, err := os.Stat(filepath.Dir(socketDir)); os.IsNotExist(err) {
+			if err := os.Mkdir(filepath.Dir(socketDir), 0755); err != nil {
+				return err
+			}
+		}
+		apiURI = fmt.Sprintf("unix:%s", socketDir)
```

You are ignoring err's that are not os.IsNotExist()